tests/integration/test_interface.py

```diff
@@ -61,40 +68,114 @@ def test_generate_dataset(
     else:
         source = _generate_non_default_data_root(dataset.name, tmpdir, sample_data_path, data)
 
-    noised_data = noising_function(seed=0, source=source, year=None)
-    noised_data_same_seed = noising_function(seed=0, source=source, year=None)
-    noised_data_different_seed = noising_function(seed=1, source=source, year=None)
+    # Update unit number cell probabilities because there are so few of them
+    # that nothing gets noised with default levels
+    config = {
+        dataset.name: {
+            Keys.COLUMN_NOISE: {
+                column.name: {
+                    noise_type.name: {
+                        Keys.CELL_PROBABILITY: 0.25,
+                    }
+                    for noise_type in COLUMNS.unit_number.noise_types
+                }
+                for column in dataset.columns
+                if "unit_number" in column.name
+            },
+        },
+    }
+
+    # Update SSA dataset to noise 'ssn' but NOT noise 'ssa_event_type' since that
+    # will be used as an identifier along with simulant_id
+    # TODO: Noise ssa_event_type when record IDs are implemented (MIC-4039)
+    if dataset.name == DATASETS.ssa.name:
+        config[dataset.name][Keys.COLUMN_NOISE][COLUMNS.ssn.name] = {
+            noise_type.name: {
+                Keys.CELL_PROBABILITY: 0.01,
+            }
+            for noise_type in COLUMNS.ssn.noise_types
+        }
+        config[dataset.name][Keys.COLUMN_NOISE][COLUMNS.ssa_event_type.name] = {
+            noise_type.name: {
+                Keys.CELL_PROBABILITY: 0,
+            }
+            for noise_type in COLUMNS.ssa_event_type.noise_types
+        }
+
+    # Set row-level noise to zero since we're just testing columns here.
+    config[dataset.name][Keys.ROW_NOISE] = {
```

Comment:
If row noise is not present, don't we have guarantees that the indexes will match in the case where we're using the sample data? 

Why aren't we testing row noise in this test? Do we have another test that is testing that row noise is applied?

Reply:
Good point - probably? I added this because the targeted omission stuff was failing. Haven't thought through yet a better way to do it with that.

As for row noise - that just seems like a different type of test and out of scope for this ticket. Maybe I'm wrong.

Reply:
After discussing w/ Rajan, we will go ahead with this approach and then immediately follow it up w/ a bugfix task